tuf/signed/sign.go

```diff
@@ -78,6 +82,20 @@ func Sign(service CryptoService, s *data.Signed, keys ...data.PublicKey) error {
 			// key is in the set of key IDs for which a signature has been created
```

Comment:
Non-blocking - apologies I couldn't add this comment 2 lines up - would it make sense to add a comment to the effect that we want to both clean old signatures and also validate the signatures we just produced before returning the signed object?